e2e/cypress/integration/messaging/quote_notation_spec.js

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Compact view: Markdown quotation', () => {
+    before(() => {
+        // # Login and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('M18704-Compact view: Markdown quotation', () => {
+        const message = '>Hello' + Date.now();
+
+        // # Create new DM channel with user's email
+        cy.apiGetUsers(['user-1', 'sysadmin']).then((userResponse) => {
+            const userEmailArray = [userResponse.body[1].id, userResponse.body[0].id];
+
+            cy.apiCreateDirectChannel(userEmailArray).then(() => {
+                cy.visit('/ad-1/messages/@sysadmin');
+
+                // # Post message to use
+                cy.postMessage(message);
+
+                // # Make sure we are on compact mode
+                cy.apiSaveMessageDisplayPreference('compact');
+
+                // # Reload the site to apply the changes on preferences
+                cy.visit('/ad-1/messages/@sysadmin');
+
+                cy.getLastPostId().then((postId) => {
+                    // * Check if the message is the same sent
+                    cy.get(`#postMessageText_${postId} > blockquote > p`).should('be.visible').and('have.text', message.slice(1));
+                    cy.get(`#post_${postId}`).find('.user-popover').then((userElement) => {
+                        // # Get the username bounding rect
+                        const userRect = userElement[0].getBoundingClientRect();
+                        cy.get(`#postMessageText_${postId}`).find('blockquote').then((quoteElement) => {
+                            // # Get the quote block bounding rect
+                            const blockQuoteRect = quoteElement[0].getBoundingClientRect();
+
+                            // * We check the username rect does not collide with the quote rect
+                            expect(userRect.right < blockQuoteRect.left || userRect.bottom < blockQuoteRect.top).to.equal(true);
```

Comment:
Not sure why we need to verify the second condition as it would not be true (as shown in this screenshot). 
```userRect.bottom < blockQuoteRect.top```

![Screenshot 2019-11-08 at 9 22 49 PM](https://user-images.githubusercontent.com/1429138/68490001-3e1d7f00-026e-11ea-93a3-d7587410e1b3.png)

Reply:
We are checking whether the username is left to the blockquote, or over it. The `userRect.bottom < blockQuoteRect.top` part is to check whether is over it. The problem is that, as you mentioned on your second comment, I am not accessing properly the username when it is in compact view. I will fix it and check the two options:
- When the username is to the left (first message after the other person talked)
- When the username is on top (second (or any other) message)
How does that sound?

Reply:
On commit 09b252a I did this solution. Please let me know if you think it is enough, or should I check other cases.